### PR TITLE
[AIR] HuggingFace Text Classification example

### PR DESCRIPTION
**Pull request by Yard1**
## Why are these changes needed?

Adds an e2e notebook example of running text classification with Ray AIR `HuggingFaceTrainer`.

Please let me know if the location is correct. We could put it in the docs instead, so that it can be rendered. Also, not sure if we should keep the cell output in or not, let me know what you think.

The notebook successfully ran on an Anyscale cluster with a head node and 2 GPU workers.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/ml/examples/huggingface/huggingface_text_classification.ipynb [hunk 1]
```diff
@@ -0,0 +1,1659 @@
...
+   "metadata": {},
+   "source": [
+    "This notebook is based on [an official 🤗 notebook - \"How to fine-tune a model on text classification\"](https://github.com/huggingface/notebooks/blob/6ca682955173cc9d36ffa431ddda505a048cbe80/examples/text_classification.ipynb). The main aim of this notebook is to show the process of conversion from vanilla 🤗 to Ray AIR 🤗 without changing the training logic unless necessary."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If you do not have 🤗 Datasets and 🤗 Transformers installed locally, uncomment and run the following line:"
```
Comment (maxpumperla) [inline python/ray/ml/examples/huggingface/huggingface_text_classification.ipynb:14]: I really like this tutorial! We should sprinkle in a couple of references/links, e.g. for Ray Tasks, runtime envs, Ray Data or AIR itself. Apart from not being a huge fan of the `trainer_init_per_worker` name, I think this makes for a relatively smooth workflow. 👍

**Comment by Yard1** [comment]
I will also add MLflow usage to this.

File: python/ray/ml/examples/huggingface/huggingface_text_classification.ipynb [hunk 2]
```diff
@@ -0,0 +1,1235 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#! pip install datasets transformers"
```
[the remark this answers is not in the document]
Comment (amogkam) [reply python/ray/ml/examples/huggingface/huggingface_text_classification.ipynb:23]: Also need to install mlflow here

File: python/ray/ml/examples/huggingface/huggingface_text_classification.ipynb [hunk 3]
```diff
@@ -0,0 +1,1235 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from ray.ml.train.integrations.huggingface import HuggingFaceTrainer\n",
+    "from ray.ml import RunConfig\n",
+    "from ray.tune.integration.mlflow import MLflowLoggerCallback\n",
+    "\n",
+    "trainer = HuggingFaceTrainer(\n",
+    "    trainer_init_per_worker=trainer_init_per_worker,\n",
+    "    scaling_config={\"num_workers\": 2, \"use_gpu\": True},\n",
```
Comment (amogkam) [inline python/ray/ml/examples/huggingface/huggingface_text_classification.ipynb:541]: Make the configurability of num_workers more obvious, and explain how this maps to the number of GPUs in the cluster.

**Review by richardliaw — COMMENTED** [review]
This is looking quite good! I'd like to try this out myself. 

Just reading through, generally, there's a lot of small tips in text. 

I think there are some of these can be better read inline as comments. For example, tips on how to toggle GPUs and `disable_tqdm` -- can be kept inline as a comment next to the parameter.

**Comment by richardliaw** [comment]
Another thing I want to comment on is that you should make sure you have some sort of high level roadmap at the top of the notebook (like a table of contents).

In the top of notebook, you should also explicitly call out and link to parts of the notebook that show capture the AIR usage.

Otherwise I'm just going with a flow and I don't exactly know what to look out for, or what I'm doing.

File: doc/source/ray-air/examples/huggingface_text_classification.ipynb [hunk 4]
```diff
@@ -0,0 +1,1520 @@
...
+              "          element.appendChild(docLink);\n",
+              "        }\n",
+              "      </script>\n",
+              "    </div>\n",
+              "  </div>\n",
+              "  "
+            ],
+            "text/plain": [
+              "     label     score\n",
+              "0  LABEL_1  0.998539\n",
```
Comment (amogkam) [inline doc/source/ray-air/examples/huggingface_text_classification.ipynb:1340]: Add a sentence on how to interpret LABEL_1 vs. LABEL_0?

File: doc/source/ray-air/examples/huggingface_text_classification.ipynb [hunk 5]
```diff
@@ -0,0 +1,1520 @@
...
+        "# !apt install git-lfs"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "5fr6E0e8hYbb"
+      },
+      "source": [
+        "Now, load the model and tokenizer locally, and recreate the `Trainer`:"
```
Comment (amogkam) [inline doc/source/ray-air/examples/huggingface_text_classification.ipynb:1425]:
```suggestion
        "Now, load the model and tokenizer locally, and recreate the Huggingface `Trainer`:"
```

**Review by amogkam — COMMENTED** [review]
This is great @Yard1! Content looks good to me! Just left some minor comments.

File: doc/source/ray-air/examples/huggingface_text_classification.ipynb (hunk 4 above)
Comment (Yard1) [reply doc/source/ray-air/examples/huggingface_text_classification.ipynb:1351]: It's model/task specific

**Review by richardliaw — APPROVED** [review]

**Pull request merged**